server/events/slack/slack_client.go

```
@@ -0,0 +1,52 @@
+package slack
+
+import (
+	"errors"
+
+	"github.com/nlopes/slack"
+)
+
+type Client interface {
+	PostMessage(text string) (string, error)
+}
+
+type ConcreteClient struct {
+	client  *slack.Client
+	channel string
+}
+
+func NewClient(slackToken string, channelName string) (*ConcreteClient, error) {
+	slackClient := slack.New(slackToken)
+
+	if _, err := slackClient.AuthTest(); err != nil {
+		return nil, err
+	}
+
+	// https://api.slack.com/faq
+	// 'How do I find a channel's ID if I only have its #name?'
```

I don't think you need this. It says you can just use the name: https://api.slack.com/methods/chat.postMessage#channels

---

With `chat.postMessage` you can use a channel name, but if the channel doesn't exist, it doesn't send.

I was thinking of making sure the `channel` from config exists on the creation of the slack client and failing early if it doesn't, rather than failing at `postMessage` or failing silently.

So to do that, I looked at https://api.slack.com/methods/channels.info which requires an ID. Alternatively, I can do a `postMessage` and see if it fails but that would post a message?

---

I like the idea of failing when Atlantis server starts if that channel doesn't exist. Failing early is usually better so good call.